### PR TITLE
backport: 'trivial' pr1

### PR DESCRIPTION
**Pull request by PastaPastaPasta**
This is a number of backports that merged 100% clean with no conflicts. Note I haven't even confirmed this builds yet, so if there are problems I'll likely fix them later

### CODE REVIEW

**Comment by PastaPastaPasta** [comment]
Rebased

File: test/README.md [hunk 1]
```diff
@@ -17,7 +17,8 @@ request is opened. Both sets of tests can also be run locally.
 
 # Running tests locally
 
-Build for your system first. Be sure to enable wallet, utils and daemon when you configure. Tests will not run otherwise.
+Before tests can be run locally, Bitcoin Core must be built.  See the [building instructions](/doc#building) for help.
```
Comment (UdjinM6) [inline test/README.md:20]: 36343da091f7edf7687e5ab694a8485d82cfd6c8  needs dashifying

File: doc/fuzzing.md [hunk 2]
```diff
@@ -70,3 +68,21 @@ $AFLPATH/afl-fuzz -i ${AFLIN} -o ${AFLOUT} -m52 -- test/test_dash_fuzzy
 
 You may have to change a few kernel parameters to test optimally - `afl-fuzz`
 will print an error and suggestion if so.
+
+## libFuzzer
+
+A recent version of `clang`, the address sanitizer and libFuzzer is needed (all
+found in the `compiler-rt` runtime libraries package).
+
+To build the `test/test_bitcoin_fuzzy` executable run
```
Comment (UdjinM6) [inline doc/fuzzing.md:77]: 5983a8877da4185f7b238d0cc969a9b7ef9be2f8 needs dashifying

**Review by UdjinM6 — CHANGES_REQUESTED** [review]
Looks good, 2 tiny issues

File: doc/developer-notes.md [hunk 3]
```diff
@@ -119,10 +119,17 @@ public:
...
+Coding Style (Python)
+---------------------
+
+Refer to [/test/functional/README.md#style-guidelines](/test/functional/README.md#style-guidelines).
 
-To facilitate the generation of documentation, use doxygen-compatible comment blocks for functions, methods and fields.
+Coding Style (Doxygen-compatible comments)
+------------------------------------------
+
+Bitcoin Core uses [Doxygen](http://www.doxygen.nl/) to generate its official documentation.
```
Comment (PastaPastaPasta) [inline doc/developer-notes.md:130]:
```suggestion
Dash Core uses [Doxygen](http://www.doxygen.nl/) to generate its official documentation.
```

File: doc/fuzzing.md (hunk 2 above)
Comment (PastaPastaPasta) [reply doc/fuzzing.md:77]:
```suggestion
To build the `test/test_dash_fuzzy` executable run
```

File: test/README.md (hunk 1 above)
Comment (PastaPastaPasta) [reply test/README.md:20]:
```suggestion
Before tests can be run locally, Dash Core must be built.  See the [building instructions](/doc#building) for help.
```

**Review by UdjinM6 — APPROVED** [review]
LGTM, utACK 👍

**Pull request merged**